hail/python/hail/gg/scale.py

```diff
@@ -0,0 +1,225 @@
+import abc
+from .geoms import FigureAttribute
+
+from hail.context import get_reference
+
+from .utils import categorical_strings_to_colors, continuous_nums_to_colors
+
+
+class Scale(FigureAttribute):
+    def __init__(self, aesthetic_name):
+        self.aesthetic_name = aesthetic_name
+
+    @abc.abstractmethod
+    def transform_data(self, field_expr):
+        pass
+
+    def transform_data_local(self, data, parent):
+        return data
+
+    @abc.abstractmethod
+    def is_discrete(self):
+        pass
+
+
+class PositionScale(Scale):
+    def __init__(self, aesthetic_name, name, breaks, labels):
+        super().__init__(aesthetic_name)
+        self.name = name
+        self.breaks = breaks
+        self.labels = labels
+
+    def update_axis(self, fig):
```

Comment:
this seems like its not meant to be used only by subclasses, right? Let's prefix it with an underscore to indicate that.

Reply:
It's meant to be used by `GGPlot` in order to update the axes of a plot with things like ticks, labels, etc.